mclagsyncd/mclaglink.cpp

```diff
@@ -401,276 +519,1390 @@ void MclagLink::setFdbEntry(char *msg, int msg_len)
+    mclag_sub_option_hdr_t    *op_hdr = NULL;
+    vector<FieldValueTuple>   fvVector;
+
+    /* Get MLAG ID */
+    op_hdr = reinterpret_cast<mclag_sub_option_hdr_t *>(static_cast<void *>(msg));
+    if (op_hdr->op_type != MCLAG_SUB_OPTION_TYPE_MCLAG_ID)
+    {
+        SWSS_LOG_ERROR("Invalid option type %u", op_hdr->op_type);
+    }
+    else
+    {
+        memcpy(&mlag_id, op_hdr->data, op_hdr->op_len);
+        p_mclag_tbl->del(to_string(mlag_id));
+        SWSS_LOG_NOTICE("Delete mlag %d", mlag_id);
+    }
+}
+
+/* Set local interface portisolate field enable/disable in the
+ * STATE_MCLAG_LOCAL_INTF_TABLE.
+ * Key = "interface"
+ */
+void MclagLink::setLocalIfPortIsolate(std::string mclag_if, bool is_enable)
+{
+    vector<FieldValueTuple>   fvVector;
+    std::string                    key;
+
+    /* Update MLAG Local Interface table: key = interface, value = * enable/disable */
+    key =  mclag_if;
+    fvVector.push_back(make_pair("port_isolate_peer_link", is_enable ? "true" : "false"));
+    p_mclag_local_intf_tbl->set(key, fvVector);
+    SWSS_LOG_NOTICE("Set local interface %s to %s", mclag_if.c_str(), is_enable ? "true" : "false");
+}
+
+/* Delete local interface
+ * STATE_MCLAG_LOCAL_INTF_TABLE.
+ * Key = "interface"
+ */
+void MclagLink::deleteLocalIfPortIsolate(std::string mclag_if)
+{
+    vector<FieldValueTuple>   fvVector;
+    std::string                    key;
+
+    p_mclag_local_intf_tbl->del(mclag_if);
+    SWSS_LOG_NOTICE("Delete local interface %s", mclag_if.c_str());
+}
+
+/* Set remote interface state field oper_status in the
+ * STATE_MCLAG_REMOTE_INTF_TABLE.
+ * Key = "Mclag<id>|interface"
+ */
+void MclagLink::mclagsyncdSetRemoteIfState(
+        char                      *msg,
+        size_t                    msg_len)
+{
+    int                       mlag_id = 0;
+    bool                      is_oper_up;
+    string                    lag_name;
+    char                      *cur;
+    size_t                    cur_len = 0;
+    mclag_sub_option_hdr_t    *op_hdr;
+    string                    key;
+    vector<FieldValueTuple>   fvVector;
+
+    while (cur_len < msg_len)
+    {
+        cur = msg + cur_len;
+        op_hdr = reinterpret_cast<mclag_sub_option_hdr_t *>(static_cast<void *>(cur));
+        switch(op_hdr->op_type)
+        {
+            case MCLAG_SUB_OPTION_TYPE_MCLAG_ID:
+                memcpy(&mlag_id, op_hdr->data, op_hdr->op_len);
+                break;
+
+            case MCLAG_SUB_OPTION_TYPE_MCLAG_INTF_NAME:
+                lag_name.insert(0, (const char*)op_hdr->data, op_hdr->op_len);
+                break;
+
+            case MCLAG_SUB_OPTION_TYPE_OPER_STATUS:
+                memcpy(&is_oper_up, op_hdr->data, op_hdr->op_len);
+                fvVector.push_back(
+                        make_pair("oper_status", is_oper_up ? "up" : "down"));
+                break;
+
+            default:
+                SWSS_LOG_WARN("Invalid option type %u", op_hdr->op_type);
```

Comment:
Invalid Option... continue or bail out?

Reply:
it is okay to continue. since this is option is not recognised by mclagsyncd although iccpd sends it. this is sub option within main option. next line increments the length and need to continue with next one
cur_len += (MCLAG_SUB_OPTION_HDR_LEN + op_hdr->op_len);